pydis_site/templates/events/pages/code-jams/9/rules.html

```diff
@@ -12,6 +12,15 @@
 {% block event_content %}
 <ol>
     <li><p>Your solution must use one of the approved frameworks (a list will be released soon). It is not permitted to circumvent this rule by e.g. using the approved framework as a wrapper for another framework.</p></li>
+    <li>
+        <p><strong>Your solution must only communicate through WebSockets</strong>. Any communication between the client and server (or any two server workers), must utilize WebSockets.</p>
+
+        <p>
+            Exceptions to this rule are made for resources such as databases and files, albeit excluding usage other than for storage.
+            For example, you may use PostgreSQL as a database but not its `NOTIFY` command.
+            Lastly, working with subprocesses (through stdin/stdout or <code>multiprocessing.Pool()</code>/<code>concurrent.futures.ProcessPoolExecutor()</code>) is also exempted from this rule.
```

Comment:
Pending discussion on the server, this wording feels a little clumsy. Would this be better?
```suggestion
            An exception to this rule is that communication with databases and files is allowed for accessing resources or for storage purposes. 
            For example, you may use PostgreSQL as a database but not its `NOTIFY` command.
            Working with subprocesses (through stdin/stdout or <code>multiprocessing.Pool()</code>/<code>concurrent.futures.ProcessPoolExecutor()</code>) is also allowed.
```

(If you prefer your wording then `exempted` should be changed to `exempt`)

Reply:
I was kind of going back- and forth- on something like this, but couldn't quite formulate it like you just did. I'll apply that.